### PR TITLE
[FIX][9.0] Fixes to sale_order_line demo

### PR DESCRIPTION
**Pull request by woodbrettm**
Discovered potentially volatile tests while doing 10.0 work. Fixes to those + other fixes below:
* Remove medical_prescription depends (covered by mail_thread_medical_prescription)
* Remove medical_pharmacy depends (covered by medical_prescription)
* Convert digits_compute attrs to digits
* Fix potentially volatile tests

### CODE REVIEW

File: sale_medical_prescription/tests/test_medical_pharmacy.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def setUp(self):
 
     def test_compute_verified_by_id(self):
         """ Test verified_by_id properly set to user """
+        self.pharmacy_4.is_verified = True
```
Comment (lasley) [inline sale_medical_prescription/tests/test_medical_pharmacy.py]: I'm confused why these are all needed? The test was passing & you're editing data without any effect on the tests whatsoever. This means either:
* The change is frivolous
* The test is broken

Which is it?

File: sale_medical_prescription/__openerp__.py [hunk 2]
```diff
@@ -15,8 +15,6 @@
     "depends": [
         "sale",
         "stock",
-        "medical_prescription",
```
Comment (lasley) [inline sale_medical_prescription/__openerp__.py]: This change is stylistic and not necessary, particularly given that the module has launched.

**Review by lasley — CHANGES_REQUESTED** [review]

File: sale_medical_prescription/__openerp__.py (hunk 2 above)
Comment (woodbrettm) [reply sale_medical_prescription/__openerp__.py]: Okie dokes.

File: sale_medical_prescription/tests/test_medical_pharmacy.py (hunk 1 above)
Comment (woodbrettm) [reply sale_medical_prescription/tests/test_medical_pharmacy.py]: Hmm. Well these tests were failing in 10.0, and these changes fixed them, so I thought that there may be a possibility they fail at some point in 9.0.

**Comment by woodbrettm** [comment]
Updated with some new changes. The current demo data for sale_order_line demo is raising ValidationErrors in sale_stock_medical_prescription, so that needed to be changed.

File: sale_medical_prescription/tests/test_medical_pharmacy.py (hunk 1 above)
Comment (lasley) [reply sale_medical_prescription/tests/test_medical_pharmacy.py]: Again, either the test is bad and more needs to be changed, or it is fine the way it was. Please decide and either revert or fix.

File: sale_medical_prescription/tests/test_medical_physician.py [hunk 3]
```diff
@@ -16,6 +16,7 @@ def setUp(self):
 
     def test_compute_verified_by_id(self):
         """ Test verified_by_id properly set to user """
+        self.physician_5.is_verified = True
```
Comment (lasley) [inline sale_medical_prescription/tests/test_medical_physician.py]: Fix test or revert change

File: sale_medical_prescription/tests/test_medical_prescription_order.py [hunk 4]
```diff
@@ -16,6 +16,7 @@ def setUp(self):
 
     def test_compute_verified(self):
         """ Test verify_user_id properly set to user """
+        self.rx_7.verify_method = 'doctor_phone'
```
Comment (lasley) [inline sale_medical_prescription/tests/test_medical_prescription_order.py]: Fix test or revert change

**Review by lasley — CHANGES_REQUESTED** [review]

**Comment by woodbrettm** [comment]
@lasley comments have been attended to:
* test_medical_physician change is required
* test_medical_pharmacy change is required
* test_medical_prescription_order change required additional changes to the related demo data. I also added a second test case to ensure the verify_date field is populated. Must have missed that previously.

Good to go.

**Comment by lasley** [comment]
I am still not understanding why you're saying the test changes are required. Do the tests fail without the changes? Making a change from a green test to a green test simply because you needed to do it in another branch is not a good enough reason to make a change. 

I need proof that the change is required. If you can't show me proof on this branch with a failing test, remove the changes.

**Comment by woodbrettm** [comment]
So I looked into this a bit more, and after a lot of mixed results, found that `<field name="">False</field` vs `<field name="" eval="False />` in the demo is what's the culprit. The changes I made to the tests are not needed as you've suggested, however I think that I should set those tests to assertTrue instead of assertEquals, as self.env.user.id changes if the tests are run when demo user is logged in. Not 100% sure on that though. If not, then the tests can stay the same and I just need to fix the demo data.

**Comment by woodbrettm** [comment]
@lasley I stripped out all the changes and ported them to 10.0. The remaining changes are because the order_lines are over-dispensing the rx_order_lines.

**Review by lasley — APPROVED** [review]

**Comment by lasley** [comment]
@tedsalmon please review

**Review by tedsalmon — APPROVED** [review]
LGTM 👍 .

**Comment by woodbrettm** [comment]
Ping

**Pull request merged**